lib/internal/streams/pipeline.js

```diff
@@ -205,14 +205,14 @@ function pipeline(...streams) {
               finish(err, null, true);
             });
         } else if (isIterable(ret, true)) {
+          wrap(pt, true, false, true);
           pump(ret, pt, finish);
         } else {
           throw new ERR_INVALID_RETURN_VALUE(
             'AsyncIterable or Promise', 'destination', ret);
         }
 
         ret = pt;
-        wrap(ret, true, false, true);
```

Comment:
This causes `finish` to be called with `final = true` unecessarily and failingly when ret is a function